src/import.cmd.js

```diff
@@ -37,6 +38,22 @@ export default class ImportCommand extends AbstractServerCommand {
 
   withUIRepo(value) {
     this._uiRepo = value;
+    this._uiBranch = GitUtils.DEFAULT_BRANCH;
+    try {
+      const url = new URL(value);
+      if (url.hash) {
+        // eslint-disable-next-line prefer-destructuring
+        this._uiRepo = `${url.origin}${url.pathname}`;
+        if (url.hash.length > 1) {
+          // eslint-disable-next-line prefer-destructuring
+          this._uiBranch = url.hash.substring(1);
+        }
+      }
```

Comment:
looks complicated, why not:

```suggestion
      if (url.hash) {
        this._uiBranch = url.hash.substring(1);
        url.hash = '';
        this._uiRepo = url.href;
      }
```

Reply:
In the case where a user provided a dangling # (with no branch), the UI Repo was invalid, so I do lines 47 & 48 every time, just in case.